frontend/src/metabase/collections/components/CollectionSidesheet/CollectionSidesheet.tsx

```diff
@@ -0,0 +1,134 @@
+import { useCallback, useMemo, useState } from "react";
+import { useMount } from "react-use";
+import { t } from "ttag";
+
+import { Sidesheet, SidesheetCard } from "metabase/common/components/Sidesheet";
+import { EntityIdCard } from "metabase/components/EntityIdCard";
+import EditableText from "metabase/core/components/EditableText";
+import { color } from "metabase/lib/colors";
+import type { ObjectWithModel } from "metabase/lib/icon";
+import { PLUGIN_COLLECTIONS } from "metabase/plugins";
+import {
+  Box,
+  Group,
+  FixedSizeIcon as Icon,
+  Stack,
+  Text,
+  Title,
+} from "metabase/ui";
+import type { Collection } from "metabase-types/api";
+
+export const CollectionSidesheet = ({
+  onClose,
+  collection,
+  onUpdateCollection,
+}: {
+  onClose: () => void;
+  collection: Collection;
+  onUpdateCollection: (entity: Collection, values: Partial<Collection>) => void;
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  useMount(() => {
+    // this component is not rendered until it is "open"
+    // but we want to set isOpen after it mounts to get
+    // pretty animations
+    setIsOpen(true);
+  });
+
+  const iconProps = useMemo(() => {
+    const icon = PLUGIN_COLLECTIONS.getIcon({
+      ...collection,
+      model: "collection",
+    } as ObjectWithModel);
+    if (icon.color) {
+      icon.color = color(icon.color);
+    }
+    return icon;
+  }, [collection]);
```

Comment:
this should all probably be encapsulated in an enterprise component

Reply:
I think PLUGIN_COLLECTIONS.getIcon is one thing in OSS and another thing in EE, so this should work in both contexts - is an enterprise component needed?

Reply:
at least here, we only show an icon at all in an enterprise context, right?